browser/ui/views/page_info/brave_page_info_bubble_view.cc

```diff
@@ -8,35 +8,154 @@
+  if (!page_info::features::IsShowBraveShieldsInPageInfoEnabled()) {
+    return;
+  }
+
+  // Obtain a reference to the Shields tab helper and observe it.  This will
+  // allow us to customize the view based on the current Shields status.
+  auto* shields_tab_helper =
+      brave_shields::BraveShieldsTabHelper::FromWebContents(web_contents());
+  if (shields_tab_helper) {
+    shields_observation_.Observe(shields_tab_helper);
+  }
+
   CustomizeChromiumViews();
+
+  // Remove the top margin set by the parent class.
+  set_margins(gfx::Insets());
+
+  // Add the tab switcher at the top of the bubble.
+  auto tab_switcher =
+      std::make_unique<BravePageInfoTabSwitcher>(base::BindRepeating(
+          &BravePageInfoBubbleView::SwitchToTab, base::Unretained(this)));
+  tab_switcher_ = AddChildViewAt(std::move(tab_switcher), 0);
+
+  // If the PageInfo bubble was not opened directly to a subpage, then show the
+  // Shields tab first.
+  tab_switcher_->SetCurrentTab(
+      IsSiteSettingsSubpageActive() ? Tab::kSiteSettings : Tab::kShields);
+  tab_switcher_->SetShieldsEnabled(IsShieldsEnabledForWebContents());
+
+  UpdateContentVisibilityForCurrentTab();
+  SizeToContents();
 }
 
 void BravePageInfoBubbleView::CustomizeChromiumViews() {
   if (!page_info::features::IsShowBraveShieldsInPageInfoEnabled()) {
     return;
   }
 
-  // Hide the close button in the page header.
+  // Hide the close button in the page header for the main page or any subpage.
   auto* close_button =
       GetViewByID(PageInfoViewFactory::VIEW_ID_PAGE_INFO_CLOSE_BUTTON);
   if (close_button) {
     close_button->SetVisible(false);
   }
+
+  // Find the first site settings content child and set its top margin.
+  for (views::View* child : children()) {
+    if (IsSiteSettingsChildView(child)) {
+      const int top_margin = IsSiteSettingsSubpageActive() ? 16 : 8;
+      child->SetProperty(views::kMarginsKey, gfx::Insets().set_top(top_margin));
+      break;
+    }
+  }
+}
+
+void BravePageInfoBubbleView::SwitchToTab(Tab tab) {
+  CHECK(tab_switcher_);
+  tab_switcher_->SetCurrentTab(tab);
+  UpdateContentVisibilityForCurrentTab();
+  SizeToContents();
+}
+
+void BravePageInfoBubbleView::UpdateContentVisibilityForCurrentTab() {
+  CHECK(tab_switcher_);
+
+  // Show/hide Chromium page info content.
+  Tab current_tab = tab_switcher_->GetCurrentTab();
+  for (views::View* child : children()) {
+    if (IsSiteSettingsChildView(child)) {
+      child->SetVisible(current_tab == Tab::kSiteSettings);
+    }
+  }
+}
+
+bool BravePageInfoBubbleView::IsSiteSettingsChildView(views::View* view) const {
+  return view->parent() == this && view != tab_switcher_;
+}
+
+bool BravePageInfoBubbleView::IsSiteSettingsSubpageActive() const {
+  // If a back button exists in the view tree, then we known that the view is
+  // displaying a subpage.
+  auto* back_button =
+      GetViewByID(PageInfoViewFactory::VIEW_ID_PAGE_INFO_BACK_BUTTON);
+  return static_cast<bool>(back_button);
```

Comment:
I think just return w/o casting is fine. AFAIK, C++ implicitely convert pointer to boolean.